### PR TITLE
Issue 170: Full review of stencil documentation and code edits

### PR DESCRIPTION
**Pull request by jdgarciauc3m**

### CODE REVIEW

File: doc/stencil.md [hunk 1]
```diff
@@ -1,31 +1,128 @@
...
+
+~~~{.cpp}
+N n = op(it1, it2, ..., itN);
+~~~
+
+The **StencilTransformer** is any C++ callable entity that takes 
+one iterator to a data item and the result of a **Neighbourhood** operation and performs
+a transformation. 
+
+Thus, a **StencilTransfomer** is any operatoio `op`that give an iterator value `it`and
```
Comment (mdolz) [inline doc/stencil.md]: Replace `operatoio` by `operation`.
Replace `give` by `gives`.

**Review by mdolz — CHANGES_REQUESTED** [review]
See comments.

**Review by Drioastorga — DISMISSED** [review]
Ok.

File: doc/stencil.md (hunk 1 above)
Comment (jdgarciauc3m) [reply doc/stencil.md]: Done

**Review by mdolz — APPROVED** [review]

**Review by javidoc00 — APPROVED** [review]
OK

**Review by Drioastorga — APPROVED** [review]

**Comment by jdgarciauc3m** [comment]
Solved issue #170

**Pull request merged**